app-backend/backsplash-core/src/main/scala/com/rasterfoundry/backsplash/MosaicImplicits.scala

```diff
@@ -113,127 +230,77 @@ class MosaicImplicits[HistStore: HistogramStore](histStore: HistStore)
               IO.raiseError(NoDataInRegionException)
             } else IO.unit
           }
-          filtered = BacksplashMosaic.filterRelevant(self)
-          extent = BacksplashImage.tmsLevels(z).mapTransform.keyToExtent(x, y)
+          overviewStream = self zip {
+            self flatMap { (backsplashIm: BacksplashImage[IO]) =>
+              fs2.Stream.eval(
+                projStore.getOverviewLocation(backsplashIm.projectLayerId)
+              )
+            }
+          } take (1) evalMap {
+            case (im, loc) =>
+              loc map { uri =>
+                IO.pure(
+                  BacksplashGeotiff(im.imageId,
+                                    im.projectId,
+                                    im.projectLayerId,
+                                    uri,
+                                    im.subsetBands,
+                                    im.corrections,
+                                    im.singleBandOptions,
+                                    im.mask,
+                                    im.footprint))
+              } getOrElse {
+                IO.raiseError(
+                  new MetadataException("No overview location found"))
```

Comment:
Would it be better if this was a more specific error than `MetadataException` so that below the `recoverWith` is more specific on its match? Just thinking that there might be error conditions that we don't know about from creating `BacksplashGeotiff` and if we're always trying to recover we're going to lose that information.

Reply:
No more error